Add nix module to Salt

KaiSforza opened the pull request:
### What does this PR do?
This module adds the ability to work with nix packages, and do some
basic operations on the nix store. There is still some work to be done,
but this lets you orchestrate some basic tasks for a system with nix
installed.

### What issues does this PR fix or reference?
None, this is a new feature.

### Tests written?
No

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html) for best practices.

KaiSforza committed:
Add nix module to Salt

This module adds the ability to work with nix packages, and do some
basic operations on the nix store. There is still some work to be done,
but this lets you orchestrate some basic tasks for a system with nix
installed.

terminalmage commented on doc/ref/modules/all/salt.modules.nix.rst:
```diff
@@ -0,0 +1,6 @@
+================
+
+.. automodule:: salt.modules.nix
+    :members:%
```
I don't think that there is supposed to be a ``%`` here.

KaiSforza replied:
You are correct, that's my bad.

terminalmage reviewed (changes requested):
This looks straightforward. Just one small issue in the RST file for the docs page.

KaiSforza committed:
nix: Check for nix-collect-garbage

I changed this from nix-store to nix-collect-garbage as the
nix.collect_garbage function used to use `nix-store --gc`, but now just
uses nix-collect-garbage.

KaiSforza committed:
nix: Correct documentation for gc

It doesn't actually show a list of all the operations, just the nix
summary.

KaiSforza commented:
@terminalmage Thanks, I removed that typo.

Also fixed a couple of problems I saw previously, as well as compacted the install output to be a single list, not a list of lists.
